server/tidb_test.go

```diff
@@ -163,6 +163,9 @@ func (ts *TidbTestSuite) TestMultiStatements(c *C) {
 func (ts *TidbTestSuite) TestSocket(c *C) {
 	cfg := config.NewConfig()
 	cfg.Socket = "/tmp/tidbtest.sock"
+	cfg.Host = ""
```

Comment:
The feature will be covered by this test if host and port is not empty.

Reply:
I did this to restore compatibility:
- Prior to this patch, specifying socket meant that the server would not listen on TCP/IP.
- Now, to prevent listening on TCP/IP you need to specify host as empty string.

I think it makes sense that the test-suite should not require TCP/IP.

Reply:
I've renamed this test to `TestSocketForwarding`, and created a new test for just `TestSocket`.